### ISSUE TITLE
优酷自频道视频无法解析了，前段时间还可以的

### ISSUE THREAD

**Issue by happyhuaer** [soimort/you-get#2004]
Please make sure these boxes are checked before submitting your issue – thank you!

- [ ] You can actually watch the video in your browser or mobile application, but not download them with `you-get`.
- [ ] Your `you-get` is up-to-date.
- [ ] I have read <https://github.com/soimort/you-get/wiki/FAQ> and tried to do so.
- [ ] The issue is not yet reported on <https://github.com/soimort/you-get/issues> or  <https://github.com/soimort/you-get/wiki/Known-Bugs>. If so, please add your comments under the existing issue.
- [ ] The issue (or question) is really about `you-get`, not about some other code or project.

Run the command with the `--debug` option, and paste the full output inside the fences:

```
[PASTE IN ME]
```

If there's anything else you would like to say (e.g. in case your issue is not about downloading a specific video; it might as well be a general discussion or proposal for a new feature), fill in the box below; otherwise, you may want to post an emoji or meme instead:

> [WRITE SOMETHING]
> [OR HAVE SOME :icecream:!]

汉语翻译最终日期：2016年02月26日

在提交前，请确保您已经检查了以下内容!

- [ ] 你可以在浏览器或移动端中观看视频，但不能使用`you-get`下载.
- [ ] 您的`you-get`为最新版.
- [ ] 我已经阅读并按 <https://github.com/soimort/you-get/wiki/FAQ> 中的指引进行了操作.
- [ ] 您的问题没有在<https://github.com/soimort/you-get/issues> , <https://github.com/soimort/you-get/wiki/FAQ> 或 <https://github.com/soimort/you-get/wiki/Known-Bugs> 报告，否则请在原有issue下报告.
- [ ] 本问题确实关于`you-get`, 而不是其他项目.

请使用`--debug`运行，并将输出粘贴在下面:

```[DEBUG] get_content: http://play.youku.com/play/get.json?vid=XMjc3NDA4OTA3Ng==&ct=10
[DEBUG] get_content: http://play.youku.com/play/get.json?vid=XMjc3NDA4OTA3Ng==&ct=12
you-get: Skipping 8 out of 8 segments due to paywall
site:                优酷 (Youku)
title:               陈亮指弹吉他进阶视频课程第40节funk练习曲详细讲解第三部分
stream:
    - format:        hd3
      container:     flv
      video-profile: 1080P
      size:          187.7 MiB (196810303 bytes)
    # download-with: you-get --format=hd3 [URL]

you-get: [Failed] Cannot extract video source.
[在这里粘贴完整日志]
```

如果您有其他附言，例如问题只在某个视频发生，或者是一般性讨论或者提出新功能，请在下面添加;或者您可以卖个萌:

> [您的内容]
> [舔 :icecream:!]

**Comment by happyhuaer** [comment]
不是的，前段时间我下载来着，可以下载，现在不行了

**Comment by rosynirvana** [comment]
或许之前没这个限制，或许之前有漏洞可以从api的层面绕过去
现在和浏览器里的行为是一致的

**Comment by lin-N** [comment]
今天也碰到这类情况，比如youku链接http://v.youku.com/v_show/id_XMjc4MjUxNzIyNA==.html
you-get版本是0.4.750
--debug出来的结果
`[DEBUG] get_content: http://play.youku.com/play/get.json?vid=XMjc4MjUxNzIyNA==&ct=10
[DEBUG] get_content: http://play.youku.com/play/get.json?vid=XMjc4MjUxNzIyNA==&ct=12
you-get: version 0.4.750, a tiny downloader that scrapes the web.
you-get: ['http://v.youku.com/v_show/id_XMjc4MjUxNzIyNA==.html']
Traceback (most recent call last):
  File "/usr/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/common.py", line 1444, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/common.py", line 1355, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/common.py", line 1157, in download_main
    download(url, **kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/common.py", line 1437, in any_download
    m.download(url, **kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/extractor.py", line 41, in download_by_url
    self.prepare(**kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/extractors/youku.py", line 206, in prepare
    self.ep = data12['security']['encrypt_string']
KeyError: 'security'`

**Comment by lin-N** [comment]
今天也碰到这类情况，比如youku链接http://v.youku.com/v_show/id_XMjc4MjUxNzIyNA==.html
you-get版本是0.4.750
--debug出来的结果
[DEBUG] get_content: http://play.youku.com/play/get.json?vid=XMjc4MjUxNzIyNA==&ct=10
[DEBUG] get_content: http://play.youku.com/play/get.json?vid=XMjc4MjUxNzIyNA==&ct=12
you-get: version 0.4.750, a tiny downloader that scrapes the web.
you-get: ['http://v.youku.com/v_show/id_XMjc4MjUxNzIyNA==.html']
Traceback (most recent call last):
  File "/usr/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/common.py", line 1444, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/common.py", line 1355, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/common.py", line 1157, in download_main
    download(url, **kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/common.py", line 1437, in any_download
    m.download(url, **kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/extractor.py", line 41, in download_by_url
    self.prepare(**kwargs)
  File "/usr/local/python34/lib/python3.4/site-packages/you_get/extractors/youku.py", line 206, in prepare
    self.ep = data12['security']['encrypt_string']
KeyError: 'security'

**Comment by rosynirvana** [comment]
估计老的ct10 ct12关了

**Comment by weinyzhou** [comment]
今天http://play.youku.com/play/get.json已经无法获取数据了{"cost":0.01600000075995922,"data":{"error":{"note":"客户端无权播放,603","code":-603}},"e":{"code":0,"provider":"hsfprovider","desc":""}}

**Comment by soimort** [comment]
Closed via #2015.
